compiler-rt/src/alu/ushl_sat.cairo

```diff
@@ -0,0 +1,62 @@
+pub mod ushl_sat_i8;
+
+use crate::utils::assert_fits_in_type;
+use crate::alu::shl::shl;
+use crate::alu::ctlz::ctlz;
+use core::num::traits::{BitSize, Bounded};
+
+// Perform the `ushl.sat` operation.
+//
+// This is a generic implementation for every data type. Its specialized versions
+// are defined and tested in the ushl_sat/ushl_sat_<type>.cairo files.
+fn ushl_sat<
```

Comment:
The same ask as for `sshl_sat` with regards to please explaining the expected requirements of the operation.